lib/Resolver/ClassResolver.php

```diff
@@ -0,0 +1,45 @@
+<?php
+declare(strict_types = 1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+
+namespace Pimcore\Resolver;
+
+/**
+ * Core class resolver returning FQCN
+ *
+ * @internal
+ */
+class ClassResolver implements ResolverInterface
+{
+    public const TYPE_DOCUMENTS = 'documents';
+    public const TYPE_ASSETS = 'assets';
+
+    private array $map;
+
+    public function __construct(array $classes = [])
+    {
+        $this->map = $classes;
+    }
+
+    public function resolve(string $class, string $type): ?string
+    {
+        if(!array_key_exists($type, $this->map)) {
+            return null;
+        }
+
+        return $this->map[$type][$class] ?? null;
```

Comment:
I don't think this is necessary, as it's already handled in the null coalescing operator in the `return` statement.

```suggestion
        return $this->map[$type][$class] ?? null;
```

Reply:
I will delete this PR any ways because it is for 11.x but we have to do it for 10.6 :) but you are right, I will put it in my 10.6 PR

Reply:
No need to delete the PR, just rebase and change the target.

Reply:
Unfortunately with all the bundle changes this is not as easy :)